nb2kg/nb2kg/managers.py

```diff
@@ -226,12 +233,10 @@ def interrupt_kernel(self, kernel_id, **kwargs):
         self.log.info("Request interrupt kernel: %s", kernel_id)
         kernel_url = self._kernel_id_to_url(kernel_id) + '/interrupt'
         self.log.info("Request restart kernel at: %s", kernel_url)
-        response = yield fetch_kg(kernel_url,
-            method='POST',
-            body=json_encode({})
-        )
-        self.log.info("Interrupt kernel response: %d %s",
-            response.code, response.reason)
+        response = yield fetch_kg(kernel_url, method='POST',
+                                  request_timeout=float(KG_REQUEST_TIMEOUT * 12),
```

Comment:
I understand increasing the request timeout by some factor, although a factor of `12` seems odd to me.  That said, regardless of the multiplier, it seems like the `restart_kernel` method should do at least the same thing, but it hasn't been modified above.
Also, we should fix the `info` log statement of (new) line 235 to reference `interrupt` rather than `restart` in the interrupt method.

Reply:
For using 12 as the multiplier, I did that according to this comment:
https://github.com/SparkTC/enterprise_gateway/issues/23#issuecomment-310223272
but yes, we need to clarify this with @ckadner the reason for choosing 12.

For `restart_kernel` we should add the `KG_REQUEST_TIMOUT` parameter to the fetch_kg method like all the others. And the `interrupt` in the log seems to be a copy-paste typo.

Another question is, do we need to do the same thing for all the methods that call `fetch_kg`? i.e. `list_kernel_specs`, `get_kernel_spec` in class `RemoteKernelSpecManager`.

Reply:
I don't have an issue with the factor, just the lack of rationalization or comment.  Not a big deal - just seems like that could use a comment.

Regarding changing all of the fetch calls, yes, it feels like the right thing to do.  However, it would be nice to come up with a scheme where only if the timeout value is "factorized" would the caller need to include it.  This will cleanup all but the `interrupt` and `restart` methods.

Not a python expert, but could this be done by adding `request_timeout=KG_REQUEST_TIMEOUT` to the parameter list of `fetch_kg()`, then the two callers that need different values, would override?

Reply:
Added a short comment for the multiplier 12.

 For the `fetch_kg()` method, here is the way I refactor it: if the two parameters `request/connect_timeout` are not provided as kwargs, we would use `KG_REQUEST/CONNECT_TIMEOUT` as default parameters value, but meanwhile it is also available to override them e.g. inside the `restart/interrupt_kernel()` methods where the request timeout could be specified by another value, e.g. increased by the multiplier 12.

```Python
def fetch_kg(endpoint, **kwargs):
    """Make an async request to kernel gateway endpoint."""
    client = AsyncHTTPClient()
    url = url_path_join(KG_URL, endpoint)

    response = yield client.fetch(url, headers=KG_HEADERS, validate_cert=VALIDATE_KG_CERT,
                                  connect_timeout=kwargs.get('connect_timeout', KG_CONNECT_TIMEOUT),
                                  request_timeout=kwargs.get('request_timeout', KG_REQUEST_TIMEOUT),
                                  auth_username=KG_HTTP_USER, auth_password=KG_HTTP_PASS, **kwargs)
    raise gen.Return(response)



def restart_kernel(self, kernel_id, now=False, **kwargs):
    ...
    response = yield fetch_kg(kernel_url, method='POST',
                                  request_timeout=float(KG_REQUEST_TIMEOUT * 12),
                                  body=json_encode({}))

```